lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2333,6 +2333,92 @@ class ConvertAtenViewOp : public OpConversionPattern<AtenViewOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenSqueezeOp : public OpConversionPattern<AtenSqueezeOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSqueezeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    Value input = adaptor.self();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    int64_t inputRank = inputType.getRank();
+    TypeConverter *typeConverter = getTypeConverter();
+    auto resultType =
+        typeConverter->convertType(op.getType()).cast<RankedTensorType>();
+    int64_t resultRank = resultType.getRank();
+
+    // In case the operand tensor type is statically shaped with all dimensions
+    // being unit extent, it will be collapsed to a 0-D tensor.
+    assert(inputRank > 0 &&
+           "expected the operand to be at lease rank 1 tensor");
+    if (resultRank == 0) {
+      SmallVector<ReassociationIndices> reassociation;
+      rewriter.replaceOpWithNewOp<linalg::TensorCollapseShapeOp>(
+          op, resultType, input, reassociation);
+      return success();
+    }
+
+    // All the static size-1 dimensions at the beginning(going from higher to
+    // lower dimensions) will be collapsed into the first dynamic or first non
+    // size-1 static dimension. All the other static size-1 dimensions will be
+    // collapsed into its previous dynamic or non size-1 static dimension.
+    SmallVector<ReassociationIndices> reassociation(resultRank);
+    bool isSqueezed = false;
+    int64_t headOnesCount = 0;
+    while (headOnesCount < inputRank && !inputType.isDynamic(headOnesCount) &&
+           inputType.getDimSize(headOnesCount) == 1) {
+      isSqueezed = true;
+      reassociation[0].push_back(headOnesCount++);
+    }
+
+    // TODO: Add support for size-1 dynamic dimensions.
+    Value one = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getIntegerAttr(rewriter.getIndexType(), 1));
+    int64_t j = -1;
```

Comment:
do you think the loop would look cleaner if arranged this way?
```
i = headOneCount;
j = 0;
while (i < inputRank && j < outputRank)
{
  while(inputType.getDimSize(i) == 1 && i < inputRank){
    reassociation[j].push_back(i);
    isSqueezed=True;
    i++;
  }
  if (inputType.isDynamicDim(i)){

  }
  reassociation[j].push_back(i);
  j++
}
check i == inputRank, j == outputRank // Both i,j should reach the end when the loop finished
```

Reply:
I think this will cause problems for trailing unit dimensions. Actually, all the unit dimensions are collapsed into the previous non unit dimension except for the unit dimensions at the beginning(which will be collapsed into the first non unit dimension).

Reply:
So reassociation for (1,2,1,1,3,1,1) would be [[0,1,2,3], [4,5,6]].